File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        ##base condition 
        if len(nums)<2: return 1
        
        L,R,cnt=0,1,1
        
        while(R<len(nums)):
            if nums[L]==nums[R]:
                R+=1
            elif nums[L]!=nums[R]:
                L+=1
                nums[L],nums[R]=nums[R],nums[L]
                cnt+=1
                R+=1
        return cnt 
 
 

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.java
class Solution {
  public int removeDuplicates(int[] nums) {
    int L=0,R=1;
    while ( R<nums.length){
    
      if(nums[L]!=nums[R]) {
        L++;
        swap(nums, L, R);
      }

      R++;
      
      
    }
    return L+1;
  }
  private void swap (int[] a, int i, int j) {
    int t = a[i];
    a[i] = a[j];
    a[j] = t;
  }
}